jans-cedarling/cedarling/src/common/cedar_schema/cedar_json.rs

```diff
@@ -62,50 +71,149 @@ impl CedarSchemaRecord {
 
 /// CedarSchemaRecordAttr defines possible type variants of the entity attribute.
 /// RecordAttr ::= STR ': {' Type [',' '"required"' ':' ( true | false )] '}'
-#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, PartialEq)]
-#[serde(untagged)]
-pub enum CedarSchemaEntityAttribute {
-    // Typed should be first to match correct fom json
-    Typed(EntityType),
-    Primitive(PrimitiveType),
+#[derive(Debug, Clone, PartialEq, serde::Serialize)]
+pub struct CedarSchemaEntityAttribute {
+    cedar_type: CedarSchemaEntityType,
+    required: bool,
 }
 
 impl CedarSchemaEntityAttribute {
     pub fn is_required(&self) -> bool {
-        match self {
-            Self::Typed(entity_type) => entity_type.required.unwrap_or(true),
-            Self::Primitive(primitive_type) => primitive_type.required.unwrap_or(true),
+        self.required
+    }
+
+    pub fn get_type(&self) -> Option<CedarType> {
```

Comment:
Should this function return `Option` or `Result` ?
Is the type mandatory ?

Reply:
You, are right. It is better to return error. 
Resolved